compiler/rustc_codegen_gcc/src/context.rs

```diff
@@ -196,15 +196,16 @@ impl<'gcc, 'tcx> CodegenCx<'gcc, 'tcx> {
 
         let mut functions = FxHashMap::default();
         let builtins = [
-            "__builtin_unreachable", "abort", "__builtin_expect", "__builtin_add_overflow", "__builtin_mul_overflow",
-            "__builtin_saddll_overflow", /*"__builtin_sadd_overflow",*/ "__builtin_smulll_overflow", /*"__builtin_smul_overflow",*/
+            "__builtin_unreachable", "abort", "__builtin_expect", /*"__builtin_expect_with_probability",*/
```

Comment:
Why was this file btw?

Reply:
@GuillaumeGomez I don't know what you are asking.

Reply:
I believe he wanted to know why you changed this file, but it seems like you only added stuff and reshuffled things, so that looks good to me.

Reply:
I added `__builtin_constant_p`. I decided to put `__builtin_constant_p` with the other non-arithmetic `__builtin*`s. Since `__builtin_expect_with_probability` was at the very end, I figured it was new and not sorted yet. I decided to put it between `__builtin_expect` and `__builtin_constant_p`. If there was a reason it was at the end I could easily move it back.

Reply:
It seems good to me as you did.